server/src/main/java/org/cloudfoundry/identity/uaa/zone/IdentityZoneEndpoints.java

```diff
@@ -336,7 +336,18 @@ public ResponseEntity<IdentityZone> deleteIdentityZone(@PathVariable String id)
             IdentityZone zone = zoneDao.retrieveIgnoreActiveFlag(id);
             // ignore the id in the body, the id in the path is the only one that matters
             IdentityZoneHolder.set(zone);
-            if (publisher != null && zone != null) {
+
+            /* reject deletion if an IdP with alias exists in the zone - checking for users with alias is not required
+             * here, since they can only exist if their origin IdP has an alias as well */
+            final List<IdentityProvider> idps = idpDao.retrieveAll(false, zone.getId());
```

Comment:
here is the breaking change.

beside this, please use a another approach to check all IdPs, see https://github.com/cloudfoundry/uaa/issues/2821 . 
If in a zone many idps installed, this can be slow.

Reply:
> please use a another approach to check all IdPs, see https://github.com/cloudfoundry/uaa/issues/2821

Was this addressed?

Reply:
It sounds like that both sides think that this should be considered as breaking change. So per our previous agreement, this should be part of major version bump. I see that it would only matter if the aliasId feature was being used. I guess that the aliasId feature was already released?

Reply:
I got further context info regarding this breaking change from other members in the team. As I mentioned earlier, "it would only matter if the aliasId feature was being used."
So when considering the scope of this PR only, because I also heard that SAP is OK including this change in non-major version bump, I see no reason to reject that from our side anymore. I think that you can phrase this change whatever way you want as needed, e.g. experimental feature, etc., maybe in release notes.
So including this in non major UAA version update is fine, but address other concerns in my other review comments.